sql/rpl_mi.cc

```diff
@@ -404,7 +404,7 @@ bool Master_info::read_info(Rpl_info_handler *from)
   if (from->get_info(&temp_master_log_pos,
                      (ulong) BIN_LOG_HEADER_SIZE) ||
       from->get_info(host, sizeof(host), (char *) 0) ||
-      from->get_info(user, sizeof(user), (char *) 0) ||
+      from->get_info(user, sizeof(user), (char *) "test") ||
```

Comment:
This change re-introduces the bug that was fixed in commit efd3968b98fd6affe5b582da34f4a7fc95c546e0.
Can you please revert this change and check if something can be fixed in test itself?